internal/v4/archive_test.go

```diff
@@ -1448,52 +1437,37 @@ func (s *ArchiveSuite) TestGetNewPromulgatedRevision(c *gc.C) {
 			Read: []string{"everyone", "tom"},
 		},
 		Promulgated: true,
-	})
-	c.Assert(err, gc.IsNil)
-	err = s.store.DB.Entities().Insert(&mongodoc.Entity{
-		URL:                 charm.MustParseReference("cs:~pluto/trusty/mariadb-5"),
-		User:                "pluto",
-		Name:                "mariadb",
-		Series:              "trusty",
-		Revision:            5,
-		PromulgatedURL:      charm.MustParseReference("cs:trusty/mariadb-3"),
-		PromulgatedRevision: 3,
-	})
-	c.Assert(err, gc.IsNil)
-	err = s.store.DB.Entities().Insert(&mongodoc.Entity{
-		URL:                 charm.MustParseReference("cs:~tom/trusty/sed-0"),
-		User:                "tom",
-		Name:                "sed",
-		Series:              "trusty",
-		Revision:            0,
-		PromulgatedURL:      charm.MustParseReference("cs:trusty/sed-0"),
-		PromulgatedRevision: 0,
-	})
-	c.Assert(err, gc.IsNil)
-	err = s.store.DB.Entities().Insert(&mongodoc.Entity{
-		URL:                 charm.MustParseReference("cs:~jerry/trusty/sed-3"),
-		User:                "jerry",
-		Name:                "sed",
-		Series:              "trusty",
-		Revision:            3,
-		PromulgatedURL:      charm.MustParseReference("cs:trusty/sed-4"),
-		PromulgatedRevision: 4,
-	})
-	c.Assert(err, gc.IsNil)
+	}}
+	for _, e := range baseEntities {
+		err := s.store.DB.BaseEntities().Insert(e)
+		c.Assert(err, gc.IsNil)
+	}
+	entities := []*mongodoc.Entity{{
+		URL:            charm.MustParseReference("cs:~pluto/trusty/mariadb-5"),
+		PromulgatedURL: charm.MustParseReference("cs:trusty/mariadb-3"),
+	}, {
+		URL:            charm.MustParseReference("cs:~tom/trusty/sed-0"),
+		PromulgatedURL: charm.MustParseReference("cs:trusty/sed-0"),
+	}, {
+		URL:            charm.MustParseReference("cs:~jerry/trusty/sed-3"),
+		PromulgatedURL: charm.MustParseReference("cs:trusty/sed-4"),
+	}}
+	for _, e := range entities {
+		charmstore.DenormalizeEntity(e)
+		err := s.store.DB.Entities().Insert(e)
+		c.Assert(err, gc.IsNil)
+	}
 	id := charm.MustParseReference("cs:~tom/trusty/awk")
 	pid := charm.MustParseReference("cs:trusty/awk")
 	for i := 0; i < 5; i++ {
 		id.Revision = i
 		pid.Revision = i
-		err = s.store.DB.Entities().Insert(&mongodoc.Entity{
-			URL:                 id,
-			User:                "tom",
-			Name:                "awk",
-			Series:              "trusty",
-			Revision:            i,
-			PromulgatedURL:      pid,
-			PromulgatedRevision: i,
-		})
+		entity := &mongodoc.Entity{
+			URL:            id,
+			PromulgatedURL: pid,
+		}
+		charmstore.DenormalizeEntity(entity)
```

Comment:
Hmm, I think this is an unreasonable usecase to export Denormalise entity for. This test should use other store methods to add the charms it needs.